azure-sfmesh-maven-plugin/src/main/java/com/microsoft/azure/maven/servicefabric/AddGatewayMojo.java

```diff
@@ -0,0 +1,146 @@
+ * license information.
+ */
+
+package com.microsoft.azure.maven.servicefabric;
+
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugin.logging.Log;
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.project.MavenProject;
+import org.codehaus.plexus.util.FileUtils;
+import org.codehaus.plexus.util.IOUtil;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+/**
+ * Goal which adds a gateway resource to a project.
+ */
+@Mojo(name = "addgateway", defaultPhase = LifecyclePhase.NONE)
+public class AddGatewayMojo extends AbstractMojo {
+
+    @Parameter(defaultValue = "${project}", required = true, readonly = true)
+    MavenProject project;
+
+    /**
+     * schema version of the gateway yaml to be generated
+    */
+    @Parameter(property = "schemaVersion", defaultValue = Constants.DEFAULT_SCHEMA_VERSION)
+    String schemaVersion;
+    
+
+    /**
+     * Name of the gateway
+    */
+    @Parameter(property = "gatewayName", required = true)
+    String gatewayName;
+
+    /**
+     * Description of the gateway
+    */
+    @Parameter(property = "gatewayDescription", defaultValue = Constants.DEFAULT_GATEWAY_DESCRIPTION)
+    String gatewayDescription;
+
+    /**
+     * Source network of gateway
+    */
+    @Parameter(property = "sourceNetwork", required = true)
+    String sourceNetwork;
+
+    /**
+     * Destination network of gateway
+    */
+    @Parameter(property = "destinationNetwork", required = true)
+    String destinationNetwork;
+
+    /**
+     * Name of the exposed endpoint
+    */
+    @Parameter(property = "tcpName", defaultValue = Constants.DEFAULT_TCP_NAME)
+    String tcpName;
+
+    /**
+     * Name of the application
+    */
+    @Parameter(property = "applicationName", required = true)
+    String applicationName;
+
+    /**
+     * Name of the service
+    */
+    @Parameter(property = "serviceName", required = true)
+    String serviceName;
+
+    /**
+     * Name of the listener
+    */
+    @Parameter(property = "listenerName", required = true)
+    String listenerName;
+    
+    /**
+     * Port to be on the listener of the service
+     */
+    @Parameter(property = "tcpPort", required = true)
+    String tcpPort;
+
+    public Log logger  = getLog();
+
+    @Override
+    public void execute() throws MojoFailureException {
+
+        final String serviceFabricResourcesDirectory = Utils.getServicefabricResourceDirectory(logger, project);
+        final String appResourcesDirectory = Utils.getAppResourcesDirectory(logger, project);
+
+        if (!Utils.checkIfExists(serviceFabricResourcesDirectory)){
+            throw new MojoFailureException("Service fabric resources folder does not exist" +
+                " Please run init goal before running this goal!");
+        } else {

```

Comment:
nit: The `else` here is not needed, since the above if condition will stop the execution by throw an exception

Reply:
changed this accordingly